src/transformers/cache_utils.py

```diff
@@ -819,15 +818,18 @@ def __init__(self, config: PretrainedConfig, max_batch_size: int, max_cache_len:
         self.key_cache: List[torch.Tensor] = []
         self.value_cache: List[torch.Tensor] = []
         cache_shape = (max_batch_size, self.num_key_value_heads, self.max_cache_len, self.head_dim)
-        for _ in range(config.num_hidden_layers):
+        for idx in range(config.num_hidden_layers):
+            # Note: `torch.export()`` requires mutations to be registered as buffers.
+            self.register_buffer(f"key_cache_{idx}", torch.zeros(cache_shape, dtype=dtype, device=device))
+            self.register_buffer(f"val_cache_{idx}", torch.zeros(cache_shape, dtype=dtype, device=device))
```

Comment:
do we have to register different buffers for each layers? Could we not just register one big buffer (num_layers, *cache_shape)?

Reply:
Addressed